subsys/pm/device_runtime.c

```diff
@@ -229,44 +225,42 @@ void pm_device_enable(const struct device *dev)
 	}
 
 out_unlock:
-	k_spin_unlock(&dev->pm->lock, key);
+	(void)k_mutex_unlock(&dev->pm->lock);
 out:
 	SYS_PORT_TRACING_FUNC_EXIT(pm, device_enable, dev);
 }
 
 void pm_device_disable(const struct device *dev)
 {
-	k_spinlock_key_t key;
-
 	SYS_PORT_TRACING_FUNC_ENTER(pm, device_disable, dev);
 	__ASSERT(k_is_pre_kernel() == false, "Device should not be disabled "
 		 "before kernel is initialized");
 
-	key = k_spin_lock(&dev->pm->lock);
+	(void)k_mutex_lock(&dev->pm->lock, K_FOREVER);
 	if (dev->pm->enable) {
 		dev->pm->enable = false;
 		/* Bring up the device before disabling the Idle PM */
 		k_work_schedule(&dev->pm->work, K_NO_WAIT);
 	}
-	k_spin_unlock(&dev->pm->lock, key);
+	(void)k_mutex_unlock(&dev->pm->lock);
 	SYS_PORT_TRACING_FUNC_EXIT(pm, device_disable, dev);
 }
 
 int pm_device_wait(const struct device *dev, k_timeout_t timeout)
 {
 	int ret = 0;
 
-	k_mutex_lock(&dev->pm->condvar_lock, K_FOREVER);
+	k_mutex_lock(&dev->pm->lock, K_FOREVER);
 	while ((k_work_delayable_is_pending(&dev->pm->work)) ||
 		(atomic_get(&dev->pm->state) == PM_DEVICE_STATE_SUSPENDING) ||
```

Comment:
we should now get rid of the atomics, `mutex` should be the responsible to protect PM data

Reply:
here is fine not using atomic, but there is one case where it is necessary. In `pm_device_request` in the sync path we need release the mutex before calling `pm_device_wait` and after this function return there is not mutex protecting the state.

But I can get rid of the atomic and use the mutex after the wait just to check the state.